public/js/test/mochitest/examples/code-simple1.js

```diff
@@ -5,3 +5,27 @@ function main() {
   const result = func();
   return result;
 }
+
+function doEval() {
+  eval("(" + function() {
+    debugger;
+
+    window.evaledFunc = function() {
+      var foo = 1;
+      var bar = 2;
+      return foo + bar;
+    };
+  }.toString() + ")()");
+}
+
+function doNamedEval() {
+  eval("(" + function() {
+    debugger;
+
+    window.evaledFunc = function() {
+      var foo = 1;
+      var bar = 2;
+      return foo + bar;
+    };
+  }.toString() + ")();\n //# sourceURL=evaled.js");
+}
```

Comment:
nit: it might be cleaner to pull out the namedEval function so that the eval is simpler.

```js
function doNamedEval() {
  const namedEval = function() {
    debugger;

    window.evaledFunc = function() {
      var foo = 1;
      var bar = 2;
      return foo + bar;
    };
  }.toString();

  eval(`(${namedEval})();\n//# sourceURL=evaled.js`);
}
```